ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -284,7 +284,8 @@ public EndpointResponse handleKsqlStatements(
               configProperties,
               localHost,
               localUrl,
-              requestConfig.getBoolean(KsqlRequestConfig.KSQL_REQUEST_INTERNAL_REQUEST)
+              requestConfig.getBoolean(KsqlRequestConfig.KSQL_REQUEST_INTERNAL_REQUEST),
+              request.getSessionVariables()
```

Comment:
Yikes, thanks for finding and fixing this bug! Unless SessionProperties (or any of its params) are mutable, it'd be good to pull out the creation of the SessionProperties to be shared between the validate() and execute() calls in this method, in order to guard against similar bugs in the future.

Reply:
One of the properties does actually get mutated unfortunately!

Reply:
Bummer (but thanks for checking)! Which one, out of curiosity?

Reply:
At some point, some items get removed from`mutableScopedProperties`